Newsweek clone

paulzay opened the pull request:
### What does this pr do?
Creates a clone project of newsweek.com
### Description of tasks to be completed?
1. Use bootstrap to create the project

paulzay committed:
Set up project

Add linters, reset.css

paulzay committed:
Update Readme file

paulzay committed:
Update Readme

paulzay committed:
add bootstrap to the project and test with some navbar template

paulzay committed:
Merge branch 'gh-pages' of https://github.com/paulzay/newsweek into gh-pages

paulzay committed:
add favicon

paulzay committed:
add all necessary import links

paulzay committed:
implement datetime feature

paulzay committed:
add sass and start working on navbar

paulzay committed:
delete styles

paulzay committed:
finish styling the navbar

paulzay committed:
start working on main categories

paulzay committed:
partition the categories using bs-grid

paulzay committed:
add cards to the featured stories container

paulzay committed:
place the top story articles

paulzay committed:
adjust the top stories grid to equal length as other grids

paulzay committed:
style the signup section, place the opinion category items

paulzay committed:
place the in-magazine items

paulzay committed:
add styling to the arts categories

paulzay committed:
finish placing the items for magazine section

paulzay committed:
place the editor's picks

paulzay committed:
place the U.S category items

paulzay committed:
place the next category sections

paulzay committed:
place the footer and start on the subscription section

paulzay committed:
style the subscribe and newsletter section

paulzay committed:
finish styling the footer

paulzay committed:
change font-size for footer linkss

paulzay committed:
style the sports section

paulzay committed:
style the other sub categories similar to sports

paulzay committed:
style the editors picks

paulzay committed:
style in the magazine section

paulzay committed:
style the top stories

paulzay committed:
style heroes and culture sections

paulzay committed:
style more stories section

paulzay committed:
style debate column

paulzay committed:
finish styling the header

paulzay committed:
fixing linter issues

paulzay committed:
more lint fixes

paulzay committed:
finish with linter fixes

paulzay committed:
fix html validation issues

paulzay committed:
style navbar for smaller screens

paulzay committed:
modify debate category order for small screens

paulzay committed:
fix lint issue

paulzay committed:
remove reduntant files

paulzay committed:
fix linter

paulzay committed:
fix bug

elmejdki commented:
### Pull Request under review by **Zakariae El Mejdki**

elmejdki commented on README.md:
```diff
@@ -1,2 +1,52 @@
+
+>USING BOOTSTRAP
+
+
```
Please add a preview picture of your project.

elmejdki commented on index.html:
```diff
@@ -0,0 +1,1532 @@
+          <li class="nav-item">
+            <span>Search</span>
+            <i class="fa fa-search"></i>
+          </li>
```
Please change this _li_ tag content into a text field and a button.

elmejdki commented on index.html:
```diff
@@ -0,0 +1,1532 @@
+            alt="firstone"
+            class="subimg1"
+          />
+          <div class="pre-card">
```
please give some padding to this class so it separate images from text.

elmejdki commented on index.html:
```diff
@@ -0,0 +1,1532 @@
+      </div>
+      <div class="row pre-foot">
+        <div class="col-12 col-lg-8 sub">
+          <img
+            src="https://d.newsweek.com/en/full/1589808/may-22-2020-cover.jpg?w=170&q=95&f=96101d0d430bef83aee5a95169c9b655"
+            alt="last one"
+            class="subimg1"
+          />
+          <img
+            src="https://d.newsweek.com/en/full/1591879/may-29-june-06-2020-cover.jpg?w=170&q=95&f=56aad51c81fc710a32b2778640b7e471"
+            alt="firstone"
+            class="subimg1"
+          />
```
Make these two images overlaps over each other as on the original website.
![image](https://user-images.githubusercontent.com/34653764/83836062-5c573a80-a6ea-11ea-92cd-136131fba1f6.png)

elmejdki commented on assets/styles/style.scss:
```diff
@@ -0,0 +1,549 @@
+  }
+
+  .arts {
+    align-items: flex-end;
+    display: flex;
```
please use Bootstrap flexbox classes instead of custom CSS.

elmejdki commented on assets/styles/style.scss:
```diff
@@ -0,0 +1,549 @@
+    }
+
+    .ft-pad {
+      border: none !important;
+
+      .card-body {
+        padding: 10px 0 !important;
```
please remove all !important keywords from your CSS, cause it is a really bad practice to override CSS with it.

elmejdki commented on index.html:
```diff
@@ -0,0 +1,1532 @@
+            x="0px"
+            y="0px"
+            viewBox="0 0 395.6 60"
+            style="enable-background: new 0 0 395.6 60;"
```
please remove all inline-styling from your HTML use your CSS files instead.

BrittanyBlake replied:
As requested by the previous TSE, please remove in-line styling from your HTML file

elmejdki reviewed (changes requested):
# Great Work @paulzay :tophat: you need only a few changes to make your project approved :smiley:.
### status: require changes

- give your web page a fixed width so it shows well on big screens like the original one.
- Make sure that you align the search input field with the menu items and also the button.
![image](https://user-images.githubusercontent.com/34653764/83834707-deddfb00-a6e6-11ea-9746-a5843c0c9146.png)

- Don't allow the main navbar links to break on small screens.
![image](https://user-images.githubusercontent.com/34653764/83834772-13ea4d80-a6e7-11ea-9f5e-cf58b00a9c40.png)

- Fix the pre-foot div (on the bottom) it gives a horizontal scroll on different breakpoints.
- Fix this overlapping section issues on tablet view:
![image](https://user-images.githubusercontent.com/34653764/83835757-8b20e100-a6e9-11ea-81c6-e90e59451a49.png)
- please add some margin between Images:
![image](https://user-images.githubusercontent.com/34653764/83835903-e94dc400-a6e9-11ea-8f4e-f5d2a91c58e6.png)

- Fix the footer to match the original page on mobile view:
<img src="https://user-images.githubusercontent.com/34653764/83837427-477ca600-a6ee-11ea-9133-7d190336548e.png" width="200" />


### kindly fix all the comments above and below before you submit an other code review.
_If you need anything don't hesitate to contact me on slack, you will find me as **Zakaraie El Mejdki**_

paulzay committed:
implement overlap images feature

paulzay committed:
change navbar li link to input

paulzay committed:
fix padding issue for the us,sports etc sections

paulzay committed:
fix overlapping image

paulzay committed:
fixing footer

paulzay committed:
fix navbar responsiveness

paulzay committed:
fix prefoot subscription section

paulzay committed:
fix linter

paulzay committed:
add screenshots and fix linter

paulzay committed:
add screenshot to readme file

BrittanyBlake reviewed (changes requested):
### Status: Changes Required ♻️
****
### Hi @paulzay !!
You have done a great job with this project so far!👍 It looks almost identical to the original and you have done a great job implementing the changes required by the TSE. 🎖️ However, there is one area where you could improve ⬇️
****

- [ ] As requested by the previous TSE, you should give your webpage a fixed width so it shows well on larger screen sizes. I have attached a screenshot of what it currently looks like on my screen 👇
![Screen Shot 2020-06-07 at 6 53 53 PM](https://user-images.githubusercontent.com/48804258/83976387-e3c6c880-a8f9-11ea-91de-c2860ff96fec.png)

- [ ] The footer is not the same as the footer on the original page on the small screen size. This was also requested by the previous TSE. I have attached screenshots of your clone as well as the original so that you can compare both.
Your clone 👇 
![Screen Shot 2020-06-07 at 8 06 08 PM](https://user-images.githubusercontent.com/48804258/83976562-f097ec00-a8fa-11ea-89dc-d42961f0dde4.png)
The original page 👇 
![Screen Shot 2020-06-07 at 8 06 30 PM](https://user-images.githubusercontent.com/48804258/83976563-f1c91900-a8fa-11ea-8fee-42e9d0fca876.png)
****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

paulzay committed:
fix footer and remove inline css

paulzay committed:
fix overflow issue and add maxwidth

paulzay committed:
fix linter

paulzay committed:
fix linter eror

HADDADSOHAIB reviewed (changes requested):
> ### Pull Request under the review by Sohaib Haddad. (TSE)
### Change Required :recycle: 

### Hello @paulzay   :handshake: 

### :medal_sports: You are doing a great job :+1:  nice work so far, it looks like the original.

### There some necessary changes before we could approve the project:
- Please implement what @BrittanyBlake asked you to do.
- Your project at very big resolution:
![Screenshot from 2020-06-08 11-14-46](https://user-images.githubusercontent.com/26064042/84019971-f1bc2e00-a979-11ea-87e1-fa90c57b6b93.png)
- The original project at big resolution:
![Screenshot from 2020-06-08 11-20-33](https://user-images.githubusercontent.com/26064042/84020058-1b755500-a97a-11ea-81f9-3a31568d6008.png)
- Also please fix this part and make it as the original:
your project:
![Screenshot from 2020-06-08 11-16-27](https://user-images.githubusercontent.com/26064042/84020107-3647c980-a97a-11ea-8a5b-52ad752a54a3.png)
The original:
![Screenshot from 2020-06-08 11-22-05](https://user-images.githubusercontent.com/26064042/84020195-56778880-a97a-11ea-9b80-3b1984c065de.png)
- Please fix validators errors:
![Screenshot from 2020-06-08 11-22-54](https://user-images.githubusercontent.com/26064042/84020246-6e4f0c80-a97a-11ea-8bc9-f17fadab29e0.png)
### Once you are done, submit for another code review. Good Luck :+1:

paulzay committed:
add fixed width for large screens and fix the newsletter section

paulzay committed:
validate the html

santiago-dev96 commented:
# Pull Request under review by @santiago-rodrig (TSE) 🧐

santiago-dev96 reviewed (changes requested):
# Changes required 🔧 

1. There is a little detail with the JS (that you shouldn't be using)

<img width="501" alt="Screen Shot 2020-06-08 at 7 09 25 AM" src="https://user-images.githubusercontent.com/43162770/84023980-0c30e000-a957-11ea-8b1b-e00f91cf3f46.png">

2. This section doesn't look similar to the one displayed on the original page (mobile screens)

**Original**

<img width="390" alt="Screen Shot 2020-06-08 at 7 14 43 AM" src="https://user-images.githubusercontent.com/43162770/84024487-fa9c0800-a957-11ea-814e-528d4dc8a43b.png">

**Yours**

<img width="361" alt="Screen Shot 2020-06-08 at 7 14 54 AM" src="https://user-images.githubusercontent.com/43162770/84024521-08518d80-a958-11ea-93a8-539b15d24ad5.png">

3. The footer social network links are too far away from the bottom line (mobile screens)

**Original**

<img width="305" alt="Screen Shot 2020-06-08 at 7 15 21 AM" src="https://user-images.githubusercontent.com/43162770/84024604-2b7c3d00-a958-11ea-92d5-2c32776bd58b.png">

**Yours** (the line is not right below the links, there is a significant separation)

<img width="282" alt="Screen Shot 2020-06-08 at 7 15 16 AM" src="https://user-images.githubusercontent.com/43162770/84024622-3767ff00-a958-11ea-94ae-97228adff7ce.png">

4. The **Start your day with our top 5 articles** section has too much width, and the heading's font-weight is too low (tablet screens)

**Original**

<img width="370" alt="Screen Shot 2020-06-08 at 7 19 56 AM" src="https://user-images.githubusercontent.com/43162770/84024913-b2311a00-a958-11ea-8553-21ba27b67095.png">

**Yours**

<img width="688" alt="Screen Shot 2020-06-08 at 7 19 49 AM" src="https://user-images.githubusercontent.com/43162770/84024926-b9582800-a958-11ea-9943-0f8bef18eb36.png">

## What to do now?

Implement the changes, and submit another CRR (Code Review Request)

## How to reach out to me?

I'm available in slack as **srodrig**, happy coding! :wave:

paulzay committed:
Implement review feedback

1. Remove js file
2. Fix the signup box and give it a maxwidth of 400px
3. Reduce the gap between social media links and underline
4. Fix the overlapping images in subscribe section

paulzay committed:
fix linter issues

HADDADSOHAIB reviewed (approved):
> ### Pull Request under the review by Sohaib Haddad. (TSE)
### Project Approved :100: 

### Hello @paulzay    :handshake: 

### Good job :clap:  your project satisfy the requirements :medal_sports: and I approve this project

### Congratulations, you can move on and merge this branch.
